java-checks/src/main/java/org/sonar/java/checks/ObjectFinalizeOverridenCallsSuperFinalizeCheck.java

```diff
@@ -113,13 +113,29 @@ private static boolean isLastStatement(@Nullable BlockTree blockTree, MethodInvo
     return false;
   }
 
-  private static boolean isFinalize(MethodTree methodTree) {
-    if ("finalize".equals(methodTree.simpleName().name()) && methodTree.parameters().isEmpty()) {
-      Tree returnType = methodTree.returnType();
-      if (returnType != null && returnType.is(Tree.Kind.PRIMITIVE_TYPE)) {
-        return "void".equals(((PrimitiveTypeTree) returnType).keyword().text());
+  private boolean isFinalize(Symbol symbol) {
+    if ("finalize".equals(symbol.name()) && symbol.isMethodSymbol()) {
+      Symbol.MethodSymbol methodSymbol = (Symbol.MethodSymbol) symbol;
+      if ("void".equals(methodSymbol.returnType().name()) && methodSymbol.parameterTypes().isEmpty()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private boolean doesOverrideFinalize(Symbol classSymbol) {
+    if (classSymbol.isTypeSymbol()) {
+      Symbol.TypeSymbol currentClass = ((Symbol.TypeSymbol) classSymbol).superClass().symbol();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'superClass' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)